### PR TITLE
Save active seconds on messages

### PR DESCRIPTION
**Pull request by nathanstitt**
Also switch to using visibility change, unload didn't seem to always fire

### CODE REVIEW

File: components/chat/chat-panel.tsx [hunk 1]
```diff
@@ -66,6 +66,8 @@ export const ChatPanel = ({
     const [transmitting, setTransmitting] = useState(false)
     const [initialMessageCount, setInitialMessageCount] = useState(0)
 
+    const { currentSeconds, ...surveyState } = useSurveyState({ user, msgCount: chat.transcript.length - initialMessageCount  })
+    console.log(currentSeconds)
```
Comment (chrisbendel) [inline components/chat/chat-panel.tsx:70]: quick fix here

File: components/chat/survey.tsx [hunk 2]
```diff
@@ -2,18 +2,20 @@ import React from 'react'
 import { styled } from '#lib/common'
 import type { ChatbotUser } from '#lib/types'
 import { Button, Drawer } from '@mantine/core'
-import { useSurveyState } from './survey-state'
+//import { useSurveyState } from './survey-state'
```
Comment (chrisbendel) [inline components/chat/survey.tsx:5]: here

File: lib/hooks.ts [hunk 3]
```diff
@@ -6,22 +6,25 @@ import { onUserEligibility } from '#server/api.telefunc'
...
-        const elapsed = now.getTime() - startTime.getTime()
+        const elapsed = now.getTime() - mountDate.getTime()
         if (elapsed > 5000) { // 5secs, only log if there's "significant time"
             cb(elapsed / 1000)
         }
-    }, [startTime, cb])
+    }, [mountDate, cb])
 
-    useEventListener('unload', fire, { target: isBrowser() ? window : null })
+    // https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon#sending_analytics_at_the_end_of_a_session
```
Comment (chrisbendel) [inline lib/hooks.ts:19]: good find on the preferred event type! good to know. what a neat little feature

**Review by chrisbendel — APPROVED** [review]

File: components/chat/chat-panel.tsx (hunk 1 above)
Comment (nathanstitt) [reply components/chat/chat-panel.tsx:70]: thanks, good catch!

**Pull request merged**